lib/performance_platform/use_case/send_performance_report.rb

```diff
@@ -0,0 +1,17 @@
+class PerformancePlatform::UseCase::SendPerformanceReport
+  def initialize(stats_gateway:, performance_gateway:)
+    @stats_gateway = stats_gateway
+    @performance_gateway = performance_gateway
+  end
+
+  def execute(presenter:)
+    stats = stats_gateway.fetch_stats
+    performance_data = presenter.present(stats: stats)
```

Comment:
On first look, this looks like it's doing two things - presenting _and_ sending stats to another gateway. Reading through to see if that's actually what's going on...

Reply:
Okay, on further look this is definitely weird to me. I don't think returning something from the `presenter` and passing it on is correct - or at least indicates that the right side of this might be a `gateway` rather than a `presenter`?

Reply:
This presenter has no side effects like a gateway does.
It takes raw stats data from the database and transforms it into something that the performance platform gateway will send across the wire.

Happy to discuss, not entirely sure I am seeing what you are seeing.